core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -481,7 +481,7 @@ object SparkSubmit extends CommandLineUtils {
         sysProp = "spark.executor.memory"),
       OptionAssigner(args.totalExecutorCores, STANDALONE | MESOS, ALL_DEPLOY_MODES,
         sysProp = "spark.cores.max"),
-      OptionAssigner(args.files, LOCAL | STANDALONE | MESOS, ALL_DEPLOY_MODES,
+      OptionAssigner(args.files, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
```

Comment:
The changes here is not correct. For yarn application, we use `spark.yarn.dist.files` to handle files, and this will be added to distributed cache. Without your change it breaks the current code.

Reply:
I met some users complained about the wired action about SparkFiles.get in yarn-client and yarn-cluster mode. SparkFiles.get is very easy for user to get file path. why not keep the same action in yarn-cluster and yarn-client mode?
Meanwhile, it not very easy for user to use spark.yarn.dist.files, it must be uploaded to HDFS in advance. To make spark on yarn more usable, use SparkFiles.get is better.

Reply:
I don't say `SparkFiles.get` is not useful, I'm saying your fix is not correct, the changes here will break the original semantics. Also we recently support remote files, to handle this scenario we should think how to address this problem for all the cluster managers, not only in yarn client mode.

Reply:
Also as @vanzin mentioned,  also the description of "Yarn cluster mode `SparkFiles.get` is working" is not a design purpose. So to fix this issue I think you should have a more solid patch.

Reply:
hi @jerryshao About the remote files to handle yarn-client files problem, is there jira that explains the design? We can wait for a version which resolved the problem.

I think my fix just solve the problem simply, do you have any other idea to solve it more elegantly?

Reply:
May I ask, why the OptionAssigner for "spark.files" works for local, standalone and mesos, only except yarn? is there any doc explain the design purpose? or may be this is really a issue.
```
OptionAssigner(args.files, LOCAL | STANDALONE | MESOS, ALL_DEPLOY_MODES, "spark.files")
```